src/components/stimuli/ClickAccuracyTest.tsx

```diff
@@ -14,15 +16,45 @@ const chartSettings = {
   width: 850,
 };
 
+interface ClickAccuracyTest {
+  distance: number;
+  speed: number;
+  clickX: number;
+  clickY: number;
+}
+
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const ClickAccuracyTest = ({ parameters, trialId }: { parameters: any, trialId: string }) => {
+const ClickAccuracyTest = ({ parameters, trialId, setAnswer }: { parameters: any, trialId: string, setAnswer: ({trialId, status, provenanceGraph, answers} : {trialId: string, status: boolean, provenanceGraph?: ProvenanceGraph<any, any, any>, answers: Record<string, any>}) => void }) => {
   const [ref, dms] = useChartDimensions(chartSettings);
   const [x, setX] = useState(100);
   const [y, setY] = useState(100);
-    const [speed, setSpeed] = useState(300);
+  const [speed, setSpeed] = useState(300);
   const flagDispatch = useFlagsDispatch();
-    const id = useLocation().pathname;
-    const taskid = parameters.taskid;
+  const id = useLocation().pathname;
+  const taskid = parameters.taskid;
+
+  const { registry, actions } = useMemo(() => {
+    const reg = Registry.create();
+
+    const click = reg.register('click', (state, click: {clickX: number, clickY: number, distance: number}) => {
+        state.clickX = click.clickX;
+        state.clickY = click.clickY;
+        state.distance = click.distance;
+        return state;
+    });
+
+    return {
+        registry: reg,
+        actions: {
+          click
+        },
+    };
+}, []);
+
+  const trrack = useMemo(() => {
+    return initializeTrrack({registry: registry, initialState: {distance: 0, speed: 10, clickX: 0, clickY: 0} });
+  }, [registry]);
```

Comment:
`useMemo` seems unnecessary here, right? Or maybe I just don't understand react well enough haha 🤷🏻

Reply:
You definitely do want a memo here I think, otherwise you'd initialize trrack on every render. But you also definitely don't need two idk why i did it like this, cleaned it up to do this in the same memo